### PR TITLE
Test Handling API Errors & Abstract Away Axios Calls

### PR DESCRIPTION
**Pull request by hyoyou**
## Description

- Use Jest `mockReturnValue()` to return a Promise object -- resolve for OK response, reject for error response
- Set up mock return value for get requests before each unit test
- Create ServerWrapper, which takes in an HTTP Client and calls methods on that client, so that Axios calls are not made directly within the components

### CODE REVIEW

File: assets/src/services/serverWrapper.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import HEADERS from '../constants/http';
+
+export default class ServerWrapper {
```
Comment (hnlee) [inline assets/src/services/serverWrapper.js:3]: Nice work!

File: assets/src/store.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
-import Vue from 'vue'
-import Vuex from 'vuex'
+import Vue from 'vue';
+import Vuex from 'vuex';
 
-Vue.use(Vuex)
+Vue.use(Vuex);
```
Comment (hnlee) [inline assets/src/store.js]: Are you actually using Vuex? Is this file necessary?

File: assets/tests/unit/home.spec.js [hunk 3]
```diff
@@ -1,12 +1,34 @@
...
+import VueRouter from 'vue-router';
 import BucketList from '@/components/BucketList.vue';
+import HEADERS from '@/constants/http.js';
 jest.mock('axios');
 import axios from 'axios';
 
 beforeEach(() => {
   jest.resetModules();
   jest.clearAllMocks();
+  let promise = new Promise(function(resolve, reject) {
```
Comment (hnlee) [inline assets/tests/unit/home.spec.js]: Might be useful to have some sort of `PromiseFactory` helper class (or a functional equivalent, depending on your preferences).

**Review by hnlee — APPROVED** [review]

File: assets/src/views/AchievedGoals.vue [hunk 4]
```diff
@@ -0,0 +1,23 @@
+<template>
+  <div class="achieved-home">
+    <AchievedList :httpClient="getHttpClient" />
+  </div>
+</template>
+
+<script>
+// @ is an alias to /src
+import AchievedList from '@/components/AchievedList.vue';
```
Comment (shibani) [inline assets/src/views/AchievedGoals.vue:9]: I like the use of @ here as an alias to the src folder... wondering if it can also be used in the following import statement `import HEADERS from '@/constants/http';` of `assets/src/services/serverWrapper.js` above?

**Review by shibani — APPROVED** [review]

File: assets/src/store.js (hunk 2 above)
Comment (hyoyou) [reply assets/src/store.js]: You're right, removed in 968b3cf!

File: assets/src/views/AchievedGoals.vue (hunk 4 above)
Comment (hyoyou) [reply assets/src/views/AchievedGoals.vue:9]: Ooh, thanks for pointing this out. I missed quite a few! Fixed in 74d8baa

**Review by hnlee — APPROVED** [review]
